finality-aleph/src/new_network/connection_manager/session.rs

```diff
@@ -94,19 +104,29 @@ impl Handler {
         if self.peers_by_node.len() + 1 == node_count {
             return Vec::new();
         }
-        (0..node_count)
-            .map(NodeIndex)
-            .filter(|node_id| *node_id != self.index() && !self.peers_by_node.contains_key(node_id))
-            .collect()
+        match self.index() {
```

Comment:
This can probably be slightly simplified by using `map_or`.

Reply:
Done